src/main/java/com/conveyal/osmlib/OSM.java

```diff
@@ -75,23 +78,41 @@ public OSM (String diskPath) {
                 // 'direct' means off-heap memory, no garbage collection overhead
                 dbMaker = DBMaker.newMemoryDirectDB(); 
             } else {
-                LOG.info("OSM will be stored in file {}.", diskPath);
-                dbMaker = DBMaker.newFileDB(new File(diskPath));
+                File dp = new File(diskPath);
+                reading = diskPath.endsWith("mapdb") && dp.exists();
```

Comment:
No problem. But is more possible that it could be mistaken for SQLite file or similar files since it is popular database [file extension](http://fileinfo.com/extension/db).